src/main/java/nl/workingtalent/wtacademy/user/UserController.java

```diff
@@ -21,76 +25,163 @@ public class UserController {
 
 	// READ
 	@RequestMapping("user/all")
-	public List<User> allUsers() {
-		return service.findAllUsers();
+	public ResponseDto findAllUsers() {
+		List<User> users = service.findAllUsers();
+		Stream<ReadUserDto> readUserDtoStream = getUsers(users);
+
+		if (users.isEmpty()) {
+			ResponseDto responseDto = new ResponseDto(false, null, null, "No users found.");
+			return responseDto;
+		}
+		ResponseDto responseDto = new ResponseDto(true, readUserDtoStream, null, users.size() + " users found.");
+		return responseDto;
 	}
 
 	@RequestMapping("user/{id}")
-	public Optional<User> findUserById(@PathVariable("id") long id) {
-		return service.findUserById(id);
+	public ResponseDto findUserById(@PathVariable("id") long id) {
+		Optional<User> userOptional = service.findUserById(id);
+
+		return createResponseDto(id, userOptional, "id");
 	}
 
 	@RequestMapping("user/firstname/{firstName}")
-	public List<User> findAllUsersByFirstName(@PathVariable("firstName") String name) {
-		return service.findUserByFirstName(name);
+	public ResponseDto findAllUsersByFirstName(@PathVariable("firstName") String name) {
+		List<User> users = service.findUserByFirstName(name);
+		Stream<ReadUserDto> readUserDtoStream = getUsers(users);
+
+		if (users.isEmpty()) {
+			ResponseDto responseDto = new ResponseDto(false, name, null,
+					"No users with first name '" + name + "' found.");
+			return responseDto;
+		}
+		ResponseDto responseDto = new ResponseDto(true, readUserDtoStream, null, users.size() + " users found.");
+		return responseDto;
 	}
 
 	@RequestMapping("user/lastname/{lastName}")
-	public List<User> findUserByLastName(@PathVariable("lastName") String name) {
-		return service.findUserByLastName(name);
+	public ResponseDto findUserByLastName(@PathVariable("lastName") String name) {
+		List<User> users = service.findUserByLastName(name);
+		Stream<ReadUserDto> readUserDtoStream = getUsers(users);
+
+		if (users.isEmpty()) {
+			ResponseDto responseDto = new ResponseDto(false, name, null,
+					"No users with last name '" + name + "' found.");
+			return responseDto;
+		}
+		ResponseDto responseDto = new ResponseDto(true, readUserDtoStream, null, users.size() + " users found.");
+		return responseDto;
```

Comment:
Je zou nog iets kunnen doen met spaties in achternamen. Bijv als ik "http://localhost:8080/user/lastname/van dijk" doe dan krijg ik wel de users, maar niet als ik "http://localhost:8080/user/lastname/vandijk" doe. Maar dit heeft geen hoge prioriteit

Reply:
Als ik op dit moment een achternaam invoer in de endpoint met een spatie krijg ik wel de user die ik zoek
<img width="1513" alt="image" src="https://github.com/Larsvandijk/WT_backend/assets/33896659/7a610437-a28e-490b-8bf9-18f7e004ce77">